### DISCUSSION TITLE
[BUG] GraphQL API: only the 10 first releases are returned instead of the 100 first

### DISCUSSION THREAD

**Discussion by ldez** [community/community#36234] [Apps, API and Webhooks] [answered]
The GitHub GraphQL API seems to bug: only the 10 first releases are returned instead of the 100 first.
`hasNextPage` is always `false` and the `endCursor` is always `MTA`.

I tried on several projects and the behavior is always the same.

This was working well for at least 2 years.

I checked:
- the changelog https://docs.github.com/en/graphql/overview/changelog
- the breaking changes https://docs.github.com/en/graphql/overview/breaking-changes

There is nothing about that, so I think it's a bug.

I tried the [GitHub GraphQL API Explorer](https://docs.github.com/en/graphql/overview/explorer)

<details>
<summary>Query</summary>

```graphql
query ($owner: String!, $name: String!, $releasesCursor:  String){
  repository(owner: $owner, name: $name) {
    releases(first: 100, after: $releasesCursor) {
      nodes {
        tagName
#        releaseAssets(first: 50) {
#          nodes {
#            downloadUrl
#          }
#        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
}
```

</details>

<details>
<summary>Variables</summary>

```graphql
{
  "owner": "golangci",
  "name": "golangci-lint",
  "releasesCursor": null
}
```

</details>

<details>
<summary>Results</summary>

```json
{
  "data": {
    "repository": {
      "releases": {
        "nodes": [
          {
            "tagName": "v1.50.0"
          },
          {
            "tagName": "v1.49.0"
          },
          {
            "tagName": "v1.48.0"
          },
          {
            "tagName": "v1.47.3"
          },
          {
            "tagName": "v1.47.2"
          },
          {
            "tagName": "v1.47.1"
          },
          {
            "tagName": "v1.47.0"
          },
          {
            "tagName": "v1.46.2"
          },
          {
            "tagName": "v1.46.1"
          },
          {
            "tagName": "v1.46.0"
          }
        ],
        "pageInfo": {
          "endCursor": "MTA",
          "hasNextPage": false
        }
      }
    }
  }
}
```

</details>

https://github.com/golangci/golangci-lint/issues/3299

**Comment by ldez** [comment]
The answer of the GitHub support:

> I thought to share that the engineering team is already looking into fixing this bug.
> 
> A temporary workaround while the team work on a solution would be to explicitly pass the order_by param in the query as that'd enable you to query the releases.
> ```
> query ($owner: String!, $name: String!, $releasesCursor:  String){
>   repository(owner: $owner, name: $name) {
>     releases(first: 100, orderBy: { field: CREATED_AT, direction: DESC }, after: $releasesCursor) {
>       nodes {
>         tagName
> #        releaseAssets(first: 50) {
> #          nodes {
> #            downloadUrl
> #          }
> #        }
>       }
>       pageInfo {
>         endCursor
>         hasNextPage
>       }
>     }
>   }
> }
> ```
> Regards,
> 
> GitHub Support

**Comment by ossanna16** [comment]
Hey there @ldez and thank you for reporting this! I'm glad you found our community 🙂

**Comment by ldez** [comment, marked as answer]
The bug has been fixed by the GitHub Team.

The answer of the GitHub support:

> I'd like to share with you that a fix has been implemented for this report. You should now be able to fetch all releases using GraphQL API without having to add the order_by parameter.
> 
> Thanks again for your report!
> 
> Regards,
> 
> GitHub Support
